### PR TITLE
Reorganize the Docs

### LINKED ISSUE #86: Tracks?
To be honest, while tracks are pretty neat, I find them to be a bit convoluted and hard to follow.  It's unlikely that most documentation readers will be patient enough to read through all of the tracks. I suggest looking at the structure of the [django documentation](https://docs.djangoproject.com/en/1.7/). It's simply split up into sections of content that is similar, as well as a tutorial at the beginning. I propose to move to a structure similar to this(potentially for end users as well?)

### PR DESCRIPTION
**Pull request by tyrannokapi**
This pull request (currently a **work-in-progress**) restructures the documentation in the following manner:

* Preparing for Development
* Creating a Plugin
  * The Basics
  * Advanced
* Contributing to Sponge
  * Sponge Contribution Guidelines
  * Coremod Development
  * API Development
  * SpongeDocs Writing
* Creating a Server
  * Getting Started
  * Server Management
  * Becoming an Expert Spongineer
* About Sponge

This structure effectively categorizes Sponge users into two groups: developers and server owners. By clearly categorizing Sponge users in this manner, we can present articles in a "basic to advanced" order without worrying too much about catering articles toward everyone. This structure will make more sense for Sponge newcomers than the current structure.

Please **do not leave a comment on every line**! I'd rather stay sane - at least for now.

Instead, please **summarize** your comments when possible.

This pull request will close #86, and potentially others.

### CODE REVIEW

**Comment by boformer** [comment]
I would say change the order to 

* Creating a Server
* Preparing for Development
* Creating a Plugin 
* Contributing to Sponge

Also, what happens to the translations when filed are renamed or moved?

**Comment by tyrannokapi** [comment]
Yes, I can do that, thanks for the suggestion.

Existing strings in the translations will remain, as far as we know. That being said, this PR likely will generate quite a few new strings once it's done.

**Comment by Tzky** [comment]
Even If this doesnt work: crowdin stores all translations in translation memory. Proofreaders can copy translation back from there. It has a Funktion for not exact matches too... Habe a look at the settins in the top right corner of the translate page.

**Comment by Spongy** [comment]
Setting up PR reference, please wait...

**Comment by tyrannokapi** [comment]
This pull request is ready for review. Note that I squashed the commits to keep things clean.

I've also edited the OP at the top of this PR for clarity. The goal of this PR was primarily to restructure the docs, not to rewrite them; however, I added/edited content when I felt it directly impacted the structure.

One change that may not be taken lightly is how I structured the FAQ, which is understandable. Recall from #133 that we discussed using the [Django documentation](https://docs.djangoproject.com/en/1.7/) as a guide for this PR. At the bottom of the Django documentation, you will find that they have a section dedicated to explaining what Django is. This is the inspiration behind adding the "About Sponge" section in this PR, which is, arguably, far easier for newcomers to understand than the current FAQ. The organization is a bit more welcoming, the length is not as daunting, and the language used is catered toward a more general audience (it is currently catered to a technically-minded audience). This approach is not perfect, and likely will not be without a fresh look at the content of the section (keep in mind that I attempted to retain the general topics discussed in the current FAQ).

Here is a visual overview of this proposed structure:
https://docs.google.com/spreadsheets/d/15ep83IfxVw7JPz0omlabLg998PjSuviK1yMaeW-1dpM/edit?usp=sharing

@BlackGoat, that Google doc may help you with #12, if this is eventually merged. :)

Note that the links in Spongy's comment seem to be outdated. Feedback is welcomed, of course. Please let me know if an article is missing, as the "99 files changed" on this PR is lower than what I counted...

**Comment by Spongy** [comment]
No sandwiches for @BlackGoat.

**Comment by BlackGoat** [comment]
This PR shuffles most of the Docs, so any concurrent PRs may need to take these changes into account. We should hold off on more PR merges until this gets through (or rejected). 
Fresh eyes were needed to hack up the bloated and aging FAQ, kudos. My feedback was via IRC, so it's mostly summarised above. Also, there is an empty "debugging" page that was originally titled "debugging Sponge in the IDE", apparently re-purposed. I haven't seen any glaring errors, does anyone else have any comments? 

Feature Completeness
Is when the Sponge API
Makes me a Sandwich

**Comment by Tzky** [comment]
Looks good so far, the condensed faq was absolutely needed as almost no one would read through this wall of text. A fresh build of this PR would help to see if the new (final?) toc looks and feels good. The google doc is fine but a bit inconvenient.

Some other PRs may need a rebase, yes. I suggest merging this first, then checking the other pending PRs.
